### PR TITLE
Add config to mark server as read-only

### PR DESCRIPTION
**Pull request by Foorack**
This PR adds a new config option to mark the server as read-only. When this option is set, the server will not allow any write operations. README and documentation have been updated to include this new option.

It can be toggled with `--read-only` or `-r` for short. Added similar tests to the existing ones to ensure that the config flag works as expected.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,10 +26,10 @@ cargo install tftpd
 tftpd --help
 ```
 
-To run the server on the IP address `0.0.0.0`, port `1234` in the `/home/user/tftp` directory:
+To run the server on the IP address `0.0.0.0`, read-only, on port `1234` in the `/home/user/tftp` directory:
 
 ```bash
-tftpd -i 0.0.0.0 -p 1234 -d "/home/user/tftp"
+tftpd -i 0.0.0.0 -p 1234 -d "/home/user/tftp" --read-only
```
Comment (altugbakan) [inline README.md:32]: Hey, thank you a lot for your addition, just a nitpick, can you use the short flag for the read-only option for this example?

```suggestion
tftpd -i 0.0.0.0 -p 1234 -d "/home/user/tftp" -r
```

**Review by altugbakan — CHANGES_REQUESTED** [review]
Thank you for adding this option, looks amazing overall!

**Comment by Foorack** [comment]
@altugbakan Thanks, suggestion applied!

**Review by altugbakan — APPROVED** [review]
LGTM

**Pull request merged**